internal/cmd/request.go

```diff
@@ -0,0 +1,104 @@
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/api/server"
+	"github.com/edulinq/autograder/internal/common"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type CommonOptions struct {
+	Verbose bool `help:"Add the full request and response to the output. Be aware that the output will include extra details beyond the expected format." default:"false"`
+}
+
+type CustomFormat func(response core.APIResponse) string
+
+func MustHandleCMDRequestAndExit(endpoint string, request any, responseType any) {
+	MustHandleCMDRequestAndExitFull(endpoint, request, responseType, CommonOptions{}, nil)
+}
+
+func MustHandleCMDRequestAndExitFull(endpoint string, request any, responseType any, options CommonOptions, customPrintFunc CustomFormat) {
+	response, err := SendCMDRequest(endpoint, request)
+	if err != nil {
+		log.Fatal("Failed to send the CMD request", err)
+	}
+
+	exitCode := PrintCMDResponseFull(request, response, responseType, options, customPrintFunc)
+
+	util.Exit(exitCode)
+}
+
+// Send a CMD request to the unix socket and return the response.
+func SendCMDRequest(endpoint string, request any) (core.APIResponse, error) {
+	socketPath, err := common.GetUnixSocketPath()
+	if err != nil {
+		return core.APIResponse{}, fmt.Errorf("Failed to get the unix socket path: '%w'.", err)
+	}
+
+	connection, err := net.Dial("unix", socketPath)
+	if err != nil {
+		return core.APIResponse{}, fmt.Errorf("Failed to dial the unix socket: '%w'", err)
+	}
+
```

Comment:
Lucas was incorrect here.
There should be no newline, since the defer is strongly related to the code above
(it is ensuring the opened connection is closed).